Runtime/SimpleGraphQL/HttpUtils.cs

```diff
@@ -21,7 +21,90 @@ public enum SubscriptionError
     [PublicAPI]
     public static class HttpUtils
     {
-        private static ClientWebSocket _webSocket;
+        public interface IWebSocket : IDisposable
+        {
+            public interface ISocketOptions
+            {
+                public void AddSubProtocol(string protocol);
+
+                public void SetRequestHeader(string v1, string v2);
+            }
+
+            WebSocketState State { get; }
+            ISocketOptions Options { get; }
+            WebSocketCloseStatus? CloseStatus { get; }
+
+            Task ConnectAsync(Uri uri, CancellationToken none);
+            Task SendAsync(ArraySegment<byte> arraySegment, WebSocketMessageType text, bool v, CancellationToken ct);
+            Task CloseAsync(WebSocketCloseStatus normalClosure, string v, CancellationToken ct);
+            Task<WebSocketReceiveResult> ReceiveAsync(ArraySegment<byte> buffer, CancellationToken ct);
+        }
+
+
+        public class ClientWebSocketOptionsWrapper : IWebSocket.ISocketOptions
+        {
+            ClientWebSocketOptions o;
+
+            public ClientWebSocketOptionsWrapper(ClientWebSocketOptions o_)
+            {
+                o = o_;
```

Comment:
`_` could be dropped from the parameter name, e.g. `this.o = o`

Reply:
Done